py/dml/codegen.py

```diff
@@ -3047,8 +3062,14 @@ def mark_method_exported(func, name, export_site):
         (otherfunc, othersite) = exported_methods[name]
         report(ENAMECOLL(export_site, othersite, name))
     else:
+        if (export_site.dml_version() != (1, 2)
+            and not func.independent):
```

Comment:
you *could* move independentness check into mark_method_statically_exported, to keep that invariant in one place

Reply:
Not convinced that's better, so I'll hold off on that unless you insist.

Reply:
that's ok